servers/rendering/shader_preprocessor.cpp

```diff
@@ -1074,8 +1081,12 @@ bool ShaderPreprocessor::expand_macros_once(const String &p_line, int p_line_num
 				}
 			}
 
-			result = result.substr(0, index) + " " + body + " " + result.substr(args_end + 1, result.length());
```

Comment:
I removed the padding " " because they made unit testing much more complicated. Have not found any case where this caused problems.

Reply:
Should be fine, like I mentioned in RocketChat adding those spaces might have been there "just in case".

Reply:
I found one case where it changes the behaviour:

```
#define MOD +
float a MOD= 1.0;
```

Previously this would've been expanded to invalid code, while now it works. I did some tests with a C++ compiler and it doesn't work there.

Should I revert this line and update the tests to require the surrounding spaces? I think it would be difficult to relax the test to allow them without adding a full blown glsl parser/tokenizer.